lib/datalog/types.proto

```diff
@@ -0,0 +1,24 @@
+syntax = "proto3";
+package datalog;
+
+// Facts represents a datalog database of predicates
+message Facts {
```

Comment:
Like mentioned above, this looks the same as access.proto above, we should only keep one.